lib/generators/action_view/component/component_generator.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Generators
+    class ComponentGenerator < Rails::Generators::NamedBase
+      source_root File.expand_path("templates", __dir__)
+
+      argument :attributes, type: :array, default: [], banner: "attribute"
+      hook_for :test_framework
+
+      def create_component_file
+        template "component.rb", File.join("app/components",  "#{file_name}.rb")
+      end
+
+      def create_template_file
+        template "component.html.erb", File.join("app/components",  "#{file_name}.html.erb")
```

Comment:
Is it possible/expected to support other view renderers here? What happens in Rails if you have Slim installed and generate a view?

Reply:
I'm not sure how we'd handle that. Is it possible to tell if the application is using Slim or Haml or any other format?

Reply:
I'm not sure.

I think it's reasonably safe to assume that ERB will be present in an application, so I'm OK with this.